src/Mobs/Silverfish.cpp

```diff
@@ -0,0 +1,66 @@
+
+#include "Globals.h"
+
+#include "Silverfish.h"
+
+#include "../World.h"
+#include "../Chunk.h"
+#include "../Blocks/BlockHandler.h"
+#include "../Blocks/BlockInfested.h"
+#include "../Entities/Player.h"
+
+bool cSilverfish::DoTakeDamage(TakeDamageInfo &a_TDI)
+{
+	bool SuperResult = Super::DoTakeDamage(a_TDI);
+	// Todo: stop this if /gamerule mobGriefing is set to false
+
+	// If the entity didn't take andy damage
+	if (!SuperResult)
+	{
+		return SuperResult;
+	}
+
+	// Entity does receive lethal damage or Attacker doesn't exist
+	if ((m_Health < a_TDI.FinalDamage) ||
+		((a_TDI.Attacker == nullptr) && (a_TDI.DamageType != dtPoison) && (a_TDI.DamageType != dtPotionOfHarming)))
+	{
+		return SuperResult;
+	}
+
+	// If attacker is player or splash potion
+	bool ShouldSpawn = false;
+	if ((a_TDI.DamageType == dtPoison) || (a_TDI.DamageType == dtPotionOfHarming))
+	{
+		ShouldSpawn = true;
+	}
+	else
+	{
+		ShouldSpawn |= a_TDI.Attacker->IsPlayer();
+	}
+
+	if (!ShouldSpawn)
+	{
+		return SuperResult;
+	}
+	auto Blocks = sSetBlockVector();
+	for (int X = static_cast<int>(m_LastPosition.x - 10); X <= static_cast<int>(m_LastPosition.x + 10); X++)
+	{
+		for (int Y = static_cast<int>(m_LastPosition.y - 5); Y <= static_cast<int>(m_LastPosition.y + 5); Y++)
+		{
+			for (int Z = static_cast<int>(m_LastPosition.z - 10); Z <= static_cast<int>(m_LastPosition.z + 10); Z++)
```

Comment:
Prefer `m_Position` over `m_LastPosition` in game logic. (`m_LastPosition` is only used for protocol reasons).